web/src/lib/InstallerClient.js

```diff
@@ -19,27 +19,39 @@
  * find current contact information at www.suse.com.
  */
 
-import axios from 'axios';
+import cockpit from './cockpit';
 
 export default class InstallerClient {
-  constructor() {
-    // this.socket = new WebSocket(`${ws}`);
-    // this.socket.onclose = () => console.log("The socket was closed");
+  // Initializing the client in the constructor does not work for some reason.
+  client() {
+    this._client = window.cockpit.dbus(
+      "org.opensuse.YaST", { bus: "system", superuser: "try" }
+    );
+    return this._client;
+  }
+
+  onPropertyChanged(handler) {
+    this.client().subscribe(
+      { interface: 'org.freedesktop.DBus.Properties', member: 'PropertiesChanged' },
+      handler
+    );
   }
 
-  onMessage(handler) {
-    this.socket.addEventListener("message", handler);
+  onSignal(signal, handler) {
+    this.client().subscribe(
+      { interface: 'org.opensuse.YaST.Installer', member: signal },
+      handler
+    )
   }
 
   authorize(username, password) {
     const auth = window.btoa(`${username}:${password}`);
 
     return new Promise((resolve, reject) => {
       return fetch(
-        "/cockpit/login", { headers: {
-          Authorization: `Basic ${auth}`,
-          "X-Superuser": "any"
-        }}).then(resp => {
+        "/cockpit/login",
+        { headers: { Authorization: `Basic ${auth}`, "X-Superuser": "any" } }
```

Comment:
Questions: why `any`? what are other valid values?

Looking for documentation, I only found 

* https://github.com/cockpit-project/cockpit/wiki/Local-sudo,-technical-design
* https://cockpit-project.org/blog/cockpit-219.html#gain-or-drop-administrative-access-in-a-running-cockpit-session
* https://github.com/cockpit-project/cockpit/blob/7d32ad6c743960a031989ea25ded80db4e7c4266/src/ws/cockpitauth.c#L919-L948
* https://github.com/cockpit-project/cockpit/blob/b71dfdef2a0aec9a58306a10205bed1e01ca0f2a/pkg/static/login.js#L553-L564